code/modules/emotes/definitions/visible.dm

```diff
@@ -358,7 +358,7 @@
 	key = "vomit"
 
 /decl/emote/visible/vomit/check_user(var/mob/living/carbon/human/user)
-	. = ..() && user.check_has_mouth() && !user.isSynthetic()
+	. = ..() && user.check_has_mouth() && user.should_have_organ(BP_STOMACH)
```

Comment:
Maybe some organs could have a list of emotes they can provide to the organ holder instead? So no need to do checks like that? For instance if a critter has a stomach, and it has the emote for vomiting listed, it's pretty safe to say there's a way for the vomit to come out and so on?